### PR TITLE
7126 - ess api add azure add b2c authentication support

### PR DESCRIPTION
**Pull request by shailen14599**
Code Changes
Functional Test case added

### CODE REVIEW

**Comment by rockydevnet** [comment]
Do we need the `Deployment/New Volume (D) - Shortcut.lnk` file?

**Comment by shailen14599** [comment]
> Do we need the `Deployment/New Volume (D) - Shortcut.lnk` file?

Hi Martin, It was added by accidentally by one of the resource. So we have identified and removed that file now.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductDataController.cs [hunk 1]
```diff
@@ -10,15 +10,14 @@
...
 using UKHO.ExchangeSetService.API.Services;
-using UKHO.ExchangeSetService.Common.Helpers;
 using UKHO.ExchangeSetService.Common.Logging;
 using UKHO.ExchangeSetService.Common.Models.Request;
 using UKHO.ExchangeSetService.Common.Models.Response;
 
 namespace UKHO.ExchangeSetService.API.Controllers
 {
     [ApiController]
-    [Authorize(Roles = ApplicationRoles.ExchangeSetServiceUser)]
```
Comment (janvi13603) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductDataController.cs:21]: Request you to remove ApplicationRoles.cs as we are no longer using it.

**Review by janvi13603 — APPROVED** [review]
One minor change requested.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductDataController.cs (hunk 1 above)
Comment (shailen14599) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductDataController.cs:21]: Resolved. Removed ApplicationRoles.cs.

**Review by martyn-fewtrell — APPROVED** [review]
Looks good to me

**Pull request merged**